src/ucp/core/ucp_worker.c

```diff
@@ -610,6 +655,7 @@ void ucp_worker_destroy(ucp_worker_h worker)
     ucs_trace_func("worker=%p", worker);
     ucp_worker_remove_am_handlers(worker);
     ucp_worker_destroy_eps(worker);
+    ucs_mpool_cleanup(&worker->am_mp, 1);
```

Comment:
it's always created in case of successful `ucp_worker_create`. As I understand, `ucp_worker_destroy` should not be called after creation failure,